drizzlepac/haputils/processing_utils.py

```diff
@@ -341,6 +344,56 @@ def compute_sregion(image, extname='SCI'):
     if closefits:
         hdu.close()
 
+def add_skycell_to_header(image, total_obj_list=None, extname='SCI'):
+    """Determines the skycells for which the image falls within and adds the 
+    information to the science header.
+
+    Parameters
+    ----------
+    image : Astropy io.fits  HDUList object
+        Image to update with the skycell keyword in each extname extension.
+    extname : str, optional
+        EXTNAME value for extension containing the header to be updated
+        
+    Returns
+    -------
+    Nothing; image header has been updated unless keyword already exists.
+    """    
+
+    # Open image and determine whether to consequently close it
+    try: 
+        hdu, closefits = _process_input(image)
+    except:
+        log.error(f"Could not open {image} during add_skycell_to_header. Exiting.")
+
+    # Find all extensions to be updated
+    numext = countExtn(hdu, extname=extname)
+    
+    for extnum in range(1, numext + 1):
+        sciext = (extname, extnum)
+        # only add skycell keyword if it does not already exist
+        if 'skycell' not in hdu[sciext].header:
+            if total_obj_list: # for SVMs
+                skycells = get_sky_cells([x.full_filename for x in total_obj_list[0].edp_list])
+            else: # for pipeline products
```

Comment:
For line 377, if the SVM contains more than one detector, then total_obj_list will be a list of 2.  As such, it looks like the exposures belonging to the second detector will not be processed.  I think SVM HST_11665_06 would be a good test case here.